src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -248,4 +261,128 @@ namespace AppInstaller::CLI::Workflow
             std::filesystem::remove(path);
         }
     }
+
+    void MSStoreInstall(Execution::Context& context)
+    {
+        // Feature toggle check
+        if (!Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::ExperimentalMSStore))
+        {
+            context.Reporter.Error() << Resource::String::FeatureDisabledMessage << std::endl;
+            AICLI_LOG(CLI, Error, << "MSStore support feature is disabled. MSStore install cancelled.");
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_EXPERIMENTAL_FEATURE_DISABLED);
+        }
+
+        auto productId = Utility::ConvertToUTF16(context.Get<Execution::Data::Installer>()->ProductId);
+
+        constexpr std::wstring_view s_StoreClientName = L"Microsoft.WindowsStore"sv;
+        constexpr std::wstring_view s_StoreClientPublisher = L"CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"sv;
+
+        // Policy check
+        AppInstallManager installManager;
+        if (installManager.IsStoreBlockedByPolicyAsync(s_StoreClientName, s_StoreClientPublisher).get())

```

Comment:
>IsStoreBlockedByPolicyAsync [](start = 27, length = 27)

Given the name of this function, I would expect that it already knows the Store's identity and that I wouldn't need to provide it.

Reply:
Yeah, unfortunately this is how the api is supposed to be called. Maybe they are thinking about multiple stores when they design this API

---
In reply to: [454000116](https://github.com/microsoft/winget-cli/pull/493#discussion_r454000116) [](ancestors = 454000116)